### ISSUE TITLE
rpi build fails

### ISSUE THREAD

**Issue by stevenjswanson** [NVSL/rpiutil#3]
[build.gtron.log.txt](https://github.com/NVSL/rpiutil/files/33701/build.gtron.log.txt)
[test.gtron.log.txt](https://github.com/NVSL/rpiutil/files/33703/test.gtron.log.txt)
[update.gtron.log.txt](https://github.com/NVSL/rpiutil/files/33702/update.gtron.log.txt)

(Gadgetron)Stevens-MacBook-Pro-4:~/research/NVSL/src/with-git/gtron_devel gtron ubt Gadgets/Tools/rpiutil/
update in Gadgets/Tools/rpiutil:                            [PASSED]
build in Gadgets/Tools/rpiutil:                             [FAILED]
WARNING: Last 10 lines of command output:
WARNING: rm -rf ../lib/libSparkFun16x2SerLCD\* libSparkFun16x2SerLCD*
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C DistanceSensor clean
echo "Cleaning up DistanceSensor"
Cleaning up DistanceSensor
rm -rf ../lib/DistanceSensor.py\* DistanceSensor.py*
rm -rf DistanceSensor_wrap*
rm -rf ../lib/_DistanceSensor.so _DistanceSensor.so
rm -rf ../lib/libDistanceSensor\* libDistanceSensor*
./shellutils/clean.sh
Command 'make' returned non-zero exit status 2
test in Gadgets/Tools/rpiutil:                              [SUCCESS]

**Comment by stevenjswanson** [comment]
Fails for another/different reason (I think ARDUPI_HOME needs to be defined.  For now, it should got in gtron_devel.sh.  you probably also need 'mkdir -p', not mkdir so it won't fail if the directory exists.):

update in Gadgets/Tools/rpiutil:                            [SUCCESS]
build in Gadgets/Tools/rpiutil:                             [FAILED]
WARNING: Last 10 lines of command output:
WARNING: echo "Cleaning up DistanceSensor"
Cleaning up DistanceSensor
rm -rf ../lib/DistanceSensor.py\* DistanceSensor.py*
rm -rf DistanceSensor_wrap*
rm -rf ../lib/_DistanceSensor.so _DistanceSensor.so
rm -rf ../lib/libDistanceSensor\* libDistanceSensor*
./shellutils/clean.sh
mkdir /lib
mkdir: /lib: Permission denied
make: **\* [all] Error 1

test in Gadgets/Tools/rpiutil:                              [SUCCESS]

**Comment by srunnikr** [comment]
Yes ARDUPI_HOME needs to be defined as I mentioned in readme. I will make
changes for mkdir, regarding the ardupi variable i need your advice. Will
the env variable in gtron_devel will do or we should do something else.
Will talk to you at the meeting today. Thanks for your input.

-sreejith

On Monday, November 16, 2015, stevenjswanson notifications@github.com
wrote:

> Fails for another/different reason (I think ARDUPI_HOME needs to be
> defined. For now, it should got in gtron_devel.sh. you probably also need
> 'mkdir -p', not mkdir so it won't fail if the directory exists.):
> 
> update in Gadgets/Tools/rpiutil: [SUCCESS]
> build in Gadgets/Tools/rpiutil: [FAILED]
> WARNING: Last 10 lines of command output:
> WARNING: echo "Cleaning up DistanceSensor"
> Cleaning up DistanceSensor
> rm -rf ../lib/DistanceSensor.py\* DistanceSensor.py*
> rm -rf DistanceSensor_wrap*
> rm -rf ../lib/_DistanceSensor.so _DistanceSensor.so
> rm -rf ../lib/libDistanceSensor\* libDistanceSensor*
> ./shellutils/clean.sh
> mkdir /lib
> mkdir: /lib: Permission denied
> make: **\* [all] Error 1
> 
> test in Gadgets/Tools/rpiutil: [SUCCESS]
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/NVSL/rpiutil/issues/3#issuecomment-157089426.

## 

Sreejith Unnikrishnan
MS CSE - UCSD - Fall 15

**Comment by stevenjswanson** [comment]
Environment variable in gtron_env.sh is the correct solution.

Your tool should build without anyone having to read the readme.

-steve

**Comment by srunnikr** [comment]
Got it. Thanks

-Sree

On Monday, November 16, 2015, stevenjswanson notifications@github.com
wrote:

> Environment variable in gtron_env.sh is the correct solution.
> 
> Your tool should build without anyone having to read the readme.
> 
> -steve
> 
> > On Nov 16, 2015, at 9:00 AM, Sreejith Unnikrishnan <
> > notifications@github.com
> > <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> > 
> > Yes ARDUPI_HOME needs to be defined as I mentioned in readme. I will make
> > changes for mkdir, regarding the ardupi variable i need your advice. Will
> > the env variable in gtron_devel will do or we should do something else.
> > Will talk to you at the meeting today. Thanks for your input.
> > 
> > -sreejith
> > 
> > On Monday, November 16, 2015, stevenjswanson <notifications@github.com
> > <javascript:_e(%7B%7D,'cvml','notifications@github.com');>>
> > wrote:
> > 
> > > Fails for another/different reason (I think ARDUPI_HOME needs to be
> > > defined. For now, it should got in gtron_devel.sh. you probably also
> > > need
> > > 'mkdir -p', not mkdir so it won't fail if the directory exists.):
> > > 
> > > update in Gadgets/Tools/rpiutil: [SUCCESS]
> > > build in Gadgets/Tools/rpiutil: [FAILED]
> > > WARNING: Last 10 lines of command output:
> > > WARNING: echo "Cleaning up DistanceSensor"
> > > Cleaning up DistanceSensor
> > > rm -rf ../lib/DistanceSensor.py\* DistanceSensor.py*
> > > rm -rf DistanceSensor_wrap*
> > > rm -rf ../lib/_DistanceSensor.so _DistanceSensor.so
> > > rm -rf ../lib/libDistanceSensor\* libDistanceSensor*
> > > ./shellutils/clean.sh
> > > mkdir /lib
> > > mkdir: /lib: Permission denied
> > > make: **\* [all] Error 1
> > > 
> > > test in Gadgets/Tools/rpiutil: [SUCCESS]
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > https://github.com/NVSL/rpiutil/issues/3#issuecomment-157089426.
> > 
> > ## 
> > 
> > Sreejith Unnikrishnan
> > MS CSE - UCSD - Fall 15
> > —
> > Reply to this email directly or view it on GitHub <
> > https://github.com/NVSL/rpiutil/issues/3#issuecomment-157097633>.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/NVSL/rpiutil/issues/3#issuecomment-157105347.

## 

Sreejith Unnikrishnan
MS CSE - UCSD - Fall 15

**Comment by BillWSY** [comment]
Still failure.

build in Gadgets/Tools/rpiutil:                             [FAILED]
WARNING: Last 10 lines of command output:
WARNING: arduPi.cpp:292:17: error: use of undeclared identifier 'CLOCK_PROCESS_CPUTIME_ID'
                clock_gettime(CLOCK_PROCESS_CPUTIME_ID, &time1);
                              ^
arduPi.cpp:295:28: error: no member named 'read' in namespace 'unistd'
                        if(available()) unistd::read(sd,&message[count],1);
                                        ~~~~~~~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
5 warnings and 20 errors generated.
make[1]: **\* [all] Error 1
make: **\* [all] Error 2
